nomt/src/io/mod.rs

```diff
@@ -59,27 +58,20 @@ impl IoKind {
         match self {
             IoKind::Read(_, _, buf) | IoKind::Write(_, _, buf) => buf,
             IoKind::WriteRaw(_, _, _, _) => panic!("attempted to extract buf from write_raw"),
-            IoKind::Fsync(_) => panic!("attempted to extract buf from fsync"),
         }
     }
 
     pub fn get_result(&self, res: i32) -> IoKindResult {
         match self {
-            // fsync returns either -1 on failure or 0 on success
-            IoKind::Fsync(_) if res == 0 => IoKindResult::Ok,
-            // pread and pwrite return the number of bytes read or written
-            IoKind::Read(_, _, _) | IoKind::Write(_, _, _) | IoKind::Write(_, _, _)
-                if res == PAGE_SIZE as i32 =>
-            {
-                IoKindResult::Ok
-            }
             // pread returns 0 if the file has been read till the end of file
             //
             // This could be a failure because the end of the file could be smaller than PAGE_SIZE.
             // However, as each read operation follows a write operation,
             // there should be no unexpected end-of-file that is not aligned with PAGE_SIZE
             // when all previous writes have succeeded.
             IoKind::Read(_, _, _) if res == 0 => IoKindResult::Ok,
+            // pread and pwrite return the number of bytes read or written
+            _ if res == PAGE_SIZE as i32 => IoKindResult::Ok,
             _ if res == -1 => IoKindResult::Err,
```

Comment:
Another thing for a follow-up:

> res holds the result of the request. Think of it like the return
value from a system call. For a normal read/write operation, this will be like the return value from read(2) or
write(2). For a successful operation, it will contain the number of bytes transferred. If a failure occurred, it will
contain the negative error value. For example, if an I/O error occurred, res will contain -EIO.

Checking for `res == -1` doesn't seem right (as you've seen in debugging this issue). Instead, we need to handle `res < 0` and turn `-res` into a std::io::Error.

Reply:
The conversion is already being done in the `linux.rs`. This method uses syscall results to make it shareable with `mod.rs`